### PR TITLE
TRUNK-5495 Replace Visit hbm mapping file with annotations

### PR DESCRIPTION
**Pull request by wluyima**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5495

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/VisitServiceTest.java [hunk 1]
```diff
@@ -938,21 +937,25 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() {
 	/**
 	 * @see VisitService#saveVisit(Visit)
 	 */
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/api/VisitServiceTest.java]: Hibernate throws this exception for the other side of the relationship i.e. if you save an encounter where visit is transient and not the opposite

File: api/src/test/java/org/openmrs/api/VisitServiceTest.java [hunk 2]
```diff
@@ -938,21 +937,25 @@ public void saveVisit_shouldAssociateEncounterWithVisitOnSaveEncounter() {
...
-	@Test(expected = TransientObjectException.class)
+	@Test
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/api/VisitServiceTest.java]: This was a bad test, hibernate throws this exception for the other side of the relationship i.e. if you save an encounter where visit is transient and not the opposite

**Comment by dkayiwa** [comment]
The link to the ticket does not look correct.

**Comment by wluyima** [comment]
Fixed it

File: api/src/test/java/org/openmrs/api/VisitServiceTest.java (hunk 2 above)
Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/VisitServiceTest.java]: If this is a bug fix, could we separate it from the switch to annotations?

Comment (wluyima) [reply api/src/test/java/org/openmrs/api/VisitServiceTest.java]: Again, this is masked by xml config, to reproduce it, the annotation changes must exist in the codebase already which will break CI (chicken-egg dilemma) which is why I have just fixed them in one PR

**Comment by wluyima** [comment]
@dkayiwa did you see my response?

**Pull request merged**